flink-core/src/test/java/org/apache/flink/types/PrimitiveDataTypeTest.java

```diff
@@ -134,106 +129,82 @@ public void testStringValue() {
         StringValue string6 = (StringValue) string0.subSequence(0, string0.length());
         StringValue string7 = (StringValue) string0.subSequence(5, 9);
         StringValue string8 = (StringValue) string0.subSequence(0, 0);
-        Assert.assertTrue(string0.compareTo(string0) == 0);
-        Assert.assertTrue(string0.compareTo(string1) == 0);
-        Assert.assertTrue(string0.compareTo(string2) > 0);
-        Assert.assertTrue(string0.compareTo(string3) < 0);
-        Assert.assertTrue(stringThis.equals(chars5));
-        Assert.assertTrue(stringThis.compareTo(string5) == 0);
-        Assert.assertTrue(string0.compareTo(string6) == 0);
-        Assert.assertTrue(stringIsA.compareTo(string7) == 0);
+        assertThat(string0.compareTo(string0)).isZero();
+        assertThat(string0.compareTo(string1)).isZero();
+        assertThat(string0.compareTo(string2)).isPositive();
+        assertThat(string0.compareTo(string3)).isNegative();
+        assertThat(chars5).isEqualTo(stringThis);
+        assertThat(stringThis.compareTo(string5)).isZero();
+        assertThat(string0.compareTo(string6)).isZero();
+        assertThat(stringIsA.compareTo(string7)).isZero();
```

Comment:
assertThatComparable(string0).isEqualByComparingTo(string0);

We can use `assertThatComparable` here.